conans/model/info.py

```diff
@@ -475,3 +476,42 @@ def default_std_non_matching(self):
 
         if self.full_settings.compiler.cppstd:
             self.settings.compiler.cppstd = self.full_settings.compiler.cppstd
+
+    def base_compatibility(self):
+        if getattr(self.full_settings.compiler, "base", None):
+            if self.full_settings.compiler.base_incompatible:
+                self.base_incompatible()
+            else:
+                self.base_compatible()
+
+    def base_compatible(self):
+        if getattr(self.full_settings.compiler, "base", None):
+            # Unfortunately assigning values is shallow
+            self.settings.compiler = (
+                self.full_settings.compiler.base
+            )  # So now self.settings.compiler is basically just a string
+
+            # Deep copy everything
+            for field, value in self.full_settings.compiler.base.as_list():
+                tokens = field.split(".")
+                attr = self.settings.compiler
+                for token in tokens[:-1]:
+                    attr = getattr(attr, token)
+                setattr(attr, tokens[-1], value)
+
+    def base_incompatible(self):
+        if getattr(self.full_settings.compiler, "base", None):
+            # Method to opt out of binary compatibility
+            self.settings.compiler = (
+                self.full_settings.compiler
+            )
+            # Set base_incompatible=True in order to make packages compatible
+            self.settings.compiler.base_incompatible = True
+
+            # Deep copy everything
+            for field, value in self.full_settings.compiler.as_list():
+                tokens = field.split(".")
+                attr = self.settings.compiler
+                for token in tokens[:-1]:
+                    attr = getattr(attr, token)
+                setattr(attr, tokens[-1], value)
```

Comment:
Both of these methods currently reset any package_id manipulation already done to the compiler. Should we try to preserve previous changes?

Reply:
yes, maybe we have to preserve those

Reply:
I am having a look at the other methods and those seem to be implemented in the same way. So I think I will leave this like this for now